### PR TITLE
cc/interval: use table-driven tests

### PR DESCRIPTION
**Pull request by jo3-l**

### CODE REVIEW

File: customcommands/interval_test.go [hunk 1]
```diff
@@ -1,315 +1,335 @@
...
+			Want:    runOnDate(day(2), hour(0), minute(0))},
+
+		{Name: "exclude all hours",
+			Cron:    "* * * * *",
+			Exclude: exclude{Hours: allHours()},
+			RefTime: now,
+			Want:    neverRuns()},
+
+		{Name: "exclude hour 0 from cron running on hour 0",
+			Cron:    "* 0 * * * *",
```
Comment (SoggySaussages) [inline customcommands/interval_test.go:233]: Extra param here?

File: customcommands/interval_test.go [hunk 2]
```diff
@@ -1,315 +1,335 @@
...
+		{Name: "on minute 5 with exclude hours",
+			Cron:    "5 * * * *",
+			Exclude: exclude{Hours: []int64{0, 1}},
+			RefTime: zero.Add(15 * time.Minute),
+			Want:    runOnDate(hour(2), minute(5))},
+
+		{Name: "on minute 0 on 2nd day with exclude hours",
+			Cron:    "0 * 2 * *",
+			Exclude: exclude{Hours: []int64{0, 1}},
+			RefTime: zero.Add(18 * time.Minute),
```
Comment (SoggySaussages) [inline customcommands/interval_test.go:253]: Why 18 here instead of 15?

File: customcommands/interval_test.go [hunk 3]
```diff
@@ -1,315 +1,335 @@
...
+
+		{Name: "on minute 5 with exclude days",
+			Cron:    "5 * * * *",
+			Exclude: exclude{Days: []time.Weekday{time.Sunday, time.Monday}},
+			RefTime: zero,
+			Want:    runOnDate(day(2), hour(0), minute(5))},
+
+		{Name: "on minute 0 on day 2,3 with exclude days",
+			Cron:    "0 * 2,3 * *",
+			Exclude: exclude{Days: []time.Weekday{time.Sunday, time.Monday}},
```
Comment (SoggySaussages) [inline customcommands/interval_test.go:276]: I believe this is supposed to exclude Tuesday with an expected date of Wednesday, I don’t think the exclude days are making a difference here. That might have been a mistake in my original.

File: customcommands/interval_test.go [hunk 4]
```diff
@@ -1,315 +1,335 @@
...
+			Cron:    "0 * 1 1,2,3,4 *",
+			Exclude: exclude{Days: []time.Weekday{time.Thursday}, Hours: allHoursExcept(1)},
+			RefTime: zero.Add(2 * time.Hour),
+			Want:    runOnDate(month(time.April), day(1), hour(1), minute(0))},
+
+		{Name: "on minute 0 of mon,tue,wed with exclude days and hours",
+			Cron:    "0 * * * 1,2,3",
+			Exclude: exclude{Days: []time.Weekday{time.Tuesday}, Hours: allHoursExcept(1)},
+			RefTime: zero.Add(2 * time.Hour),
+			Want:    runOnDate(day(2), hour(1), minute(0))},
```
Comment (SoggySaussages) [inline customcommands/interval_test.go:314]: Shouldn’t this be expected to run on Day 3?

**Review by SoggySaussages — COMMENTED** [review]
This is very cool, thank you for taking the time to make all my nonsense coherent and nice. I want to know this PR carnally.

File: customcommands/interval_test.go (hunk 4 above)
Comment (jo3-l) [reply customcommands/interval_test.go:314]: Whoops! I changed this temporarily to demonstrate the test output on failures, but forgot to change things back.

File: customcommands/interval_test.go (hunk 1 above)
Comment (jo3-l) [reply customcommands/interval_test.go:233]: Good catch. I will correct this and add some validation to ensure we don't test invalid crons shortly.

File: customcommands/interval_test.go (hunk 3 above)
Comment (jo3-l) [reply customcommands/interval_test.go:276]: Your original test was probably correct :-) It's likely I threw in an off-by-one when converting Days to `time.Weekday` values from `int64`s. Will fix.

File: customcommands/interval_test.go (hunk 2 above)
Comment (jo3-l) [reply customcommands/interval_test.go:253]: No real reason; I like to vary test inputs when I can but I agree it is unnecessarily confusing here. Will change to 15.

**Pull request merged**